File: PromptCraft-Studio/.claude/worktrees/objective-wilbur-721c52/frontend/src/utils/variableParser.ts
/** 从文本中提取 {{变量名}}，去重，保持首次出现顺序 */
export function parseVariables(text: string): string[] {
  if (!text) return []
  const regex = /\{\{(\w+)\}\}/g
  const seen = new Set<string>()
  const result: string[] = []
  let match: RegExpExecArray | null
  while ((match = regex.exec(text)) !== null) {
    if (!seen.has(match[1])) {
      seen.add(match[1])
      result.push(match[1])
    }
  }
  return result
}

File: PromptCraft-Studio/backend/src/main/java/com/promptcraft/util/VariableParser.java
package com.promptcraft.util;

import org.springframework.stereotype.Component;

import java.util.ArrayList;
import java.util.LinkedHashSet;
import java.util.List;
import java.util.regex.Matcher;
import java.util.regex.Pattern;

@Component
public class VariableParser {

    private static final Pattern VAR_PATTERN = Pattern.compile("\\{\\{(\\w+)\\}\\}");

    /** 从单段文本中提取变量名，去重，保持首次出现顺序 */
    public List<String> parseVariables(String text) {
        if (text == null || text.isBlank()) return List.of();
        LinkedHashSet<String> seen = new LinkedHashSet<>();
        Matcher m = VAR_PATTERN.matcher(text);
        while (m.find()) seen.add(m.group(1));
        return new ArrayList<>(seen);
    }

    /** 合并 systemPrompt + userPrompt 后提取变量名（systemPrompt 可为 null） */
    public List<String> parseFromPrompt(String systemPrompt, String userPrompt) {
        String combined = (systemPrompt != null ? systemPrompt : "") + " " + (userPrompt != null ? userPrompt : "");
        return parseVariables(combined);
    }
}
